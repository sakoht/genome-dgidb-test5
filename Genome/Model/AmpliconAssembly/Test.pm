package Genome::Model::AmpliconAssembly::Test;

use strict;
use warnings;

use base 'Test::Class';

use File::Copy 'copy';
use File::Temp 'tempdir';
use Genome::InstrumentData::Sanger::Test;
use Genome::Model::InstrumentDataAssignment;
use Genome::ProcessingProfile::AmpliconAssembly::Test;
use Genome::Utility::FileSystem;
use Test::More;

#< DIR >#
sub test_dir {
    return '/gsc/var/cache/testsuite/data/Genome-Model-AmpliconAssembly';
}

sub _test_dir_for_type {
    die "No type to get test dir\n" if not defined $_[0] or $_[0] eq __PACKAGE__;
    return test_dir().'/'.$_[0];
}

#< MOCK ># 
sub create_mock_model {
    my $self = shift;

    # Processing profile
    my $pp = Genome::ProcessingProfile::AmpliconAssembly::Test->create_mock_processing_profile
        or return;

    #my $data_dir = File::Temp::tempdir(DIR => $self->test_dir, CLEANUP => 0);
    my $data_dir = File::Temp::tempdir(CLEANUP => 1);
    die "Can't make temp data directory for mock model\n" unless -d $data_dir;

    # Model
    my $model = Genome::Model::AmpliconAssembly->create_mock(
        id => -5000,
        genome_model_id => -5000,
        name => 'mr. mock',
        subject_name => 'mock_dna',
        subject_type => 'dna_resource_item_name',
        processing_profile_id => $pp->id,
        processing_profile => $pp,
        data_directory => $data_dir,
    )
        or die "Can't create mock model for amplicon assembly\n";
    for my $pp_param ( Genome::ProcessingProfile::AmpliconAssembly->params_for_class ) {
        $model->set_always($pp_param, $pp->$pp_param);
    }
    for my $dir_type ( Genome::Consed::Directory->directories ) {
        my $dir = $data_dir.'/'.$dir_type; 
        $model->set_always($dir_type, $dir);
        mkdir $dir
            or die "Can't make directory ($dir): $!\n";
    }
    $model->mock(
        'amplicons',
        sub{  
            my $dh = Genome::Utility::FileSystem->open_directory( $model->chromat_dir )
                or return;

            my %amplicons;
            while ( my $scf = $dh->read ) {
                next if $scf =~ m#^\.#;
                $scf =~ s#\.gz##;
                $scf =~ /^(.+)\.[bg]\d+$/
                    or next;
                push @{$amplicons{$1}}, $scf;
            }
            $dh->close;

            return \%amplicons;
        },
    );

    #< FILES >#
    for my $type (qw/ assembly reads processed assembly.confirmed assembly.unconfirmed /) {
        my $method = $type.'_fasta';
        $model->set_always($method, sprintf('%s/%s.%s.fasta', $data_dir, $model->subject_name, $type));
    }
    $model->set_always('metrics_file',  sprintf('%s/%s.metrics.txt', $data_dir, $model->subject_name));

    return $model;
}

sub create_mock_model_with_instrument_data {
    my $self = shift;

    my $model = $self->create_mock_model
        or return;

    my $inst_data = Genome::InstrumentData::Sanger::Test->create_mock_instrument_data; # this dies if no workee
    my $ida = Genome::Model::InstrumentDataAssignment->create_mock(
        id => -5000,
        model_id => $model->id,
        instrument_data_id => $inst_data->id,
        first_build_id => undef,
    )
        or die "Can't create mock instrument data assignment\n";
    $ida->set_always('model_id', $model);
    $ida->set_always('instrument_data', $inst_data);
    $ida->mock(
        'first_build_id', sub { 
            my ($ida, $fbi) = @_;
            $ida->{first_build_id} = $fbi if defined $fbi;
            return $ida->{first_build_id}; 
        },
    );
    $model->set_always('instrument_data', ( $inst_data ));
    $model->set_always('instrument_data_assignments', ( $ida ));

    return $model;
}

#< COPY DATA >#
sub copy_dirs_to_model {
    my ($self, $model, @dirs) = @_;

    die "Need dir types to copy from\n" unless @dirs;

    for my $dir ( @dirs ) {
        my $dest = ( $model->can($dir) )
        ? $model->$dir
        : $model->data_directory;
        $self->_copy_dir(_test_dir_for_type($dir), $dest)
            or return;
    }

    return 1;
}

sub _copy_dir {
    my ($self, $source, $dest) = @_;

    my $dh = Genome::Utility::FileSystem->open_directory($source)
        or return;

    while ( my $file = $dh->read ) {
        next if $file =~ m#^\.#;
        File::Copy::copy("$source/$file", $dest)
            or die "Can't copy ($source/$file) to ($dest): $!\n";
    }

    return 1;

}

#< REAL MODEL FOR TESTING >#

1;

=pod

=head1 Name

ModuleTemplate

=head1 Synopsis

=head1 Usage

=head1 Methods

=head2 

=over

=item I<Synopsis>

=item I<Arguments>

=item I<Returns>

=back

=head1 See Also

=head1 Disclaimer

Copyright (C) 2005 - 2008 Washington University Genome Sequencing Center

This module is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> I<ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

