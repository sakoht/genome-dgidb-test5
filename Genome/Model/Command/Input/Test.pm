
###########################################################################

package Genome::Model::Command::Input::Test;

use strict;
use warnings;

use base 'Test::Class';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input';
}



###########################################################################

package Genome::Model::Command::Input::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;
require Genome::Model::Test;
require File::Temp;
require File::Path;

sub valid_param_sets {
    return grep { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub invalid_param_sets {
    return map { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::Test->create_basic_mock_model(type_name => 'tester')
            or confess "Can't create mock tester model.";
    }

    return $self->{_model};
}

sub startup : Tests(startup => 1) {
    my $self = shift;
    
    no warnings 'once';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    is_deeply(
        [ Genome::Model::Command->get_model_type_names ],
        [qw/ tester /],
        'Set tester type name.',
    );

    $self->_startup or confess "Failed startup.";
    
    return 1;
}

sub _startup {
    return 1;
}

# 
sub __pre_execute {
    my $self = shift;

    print Dumper([$self->_model->inputs]);
    
    return 1;
}

sub __post_execute {
    my $self = shift;

    print Dumper([$self->_model->inputs]);
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Add::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Add';
}

sub _valid_param_sets {
    return (
        { # add multiple 'value_ids' 
            name => 'friends',
            values => 'Watson,Crick',
        },
        { # add single 'value'
            name => 'inst_data',
            values => '2sep09.934pmaa1',
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to add a not 'is_many' input
            name => 'coolness', 
            values => 'none',
        },
        { # try to add Crick again
            name => 'friends',
            values => 'Crick',
        },
        { # try to add id that don't exist
            name => 'inst_data',
            values => '2sep09.934noexist',
        },
    );
}

sub _pre_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @values = $self->_model->$name;
    ok(!@values, "No $name found in model.");
    
    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @values = $self->_model->$name;
    ok(@values, "Added $name in model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Remove::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Remove';
}

sub _valid_param_sets {
    return (
        { # remove multiple 'value_ids' 
            name => 'friends',
            values => 'Watson,Crick',
        },
        { # remove single 'value'
            name => 'inst_data',
            values => '2sep09.934pmaa1',
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to remove a not 'is_many' input
            name => 'coolness', 
            values => 'none',
        },
        { # try to remove input that is not linked to the model anymore
            name => 'inst_data',
            values => '2sep09.934pmaa1',
        },
    );
}

sub friends {
    return 'Crick';
}

sub inst_data {
    return Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
}

sub _startup {
    my $self = shift;

    my $model = $self->_model;

    $model->add_friend('Crick');
    $model->add_friend('Watson');
    my @friends = $model->friends();
    is_deeply(\@friends, [qw/ Crick Watson /], 'Added friends to remove.');
    my $id = Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
        or confess "Can't get sange id 2sep09.934pmaa1";
    $model->add_inst_data($id);
    my @inst_data = $model->inst_data;
    is_deeply(\@inst_data, [ $id ], 'Added instr_data to remove');

    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @values = $self->_model->$name;
    #print Dumper(\@values);
    ok(!@values, "Removed $name from model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::List::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::List';
}

sub valid_param_sets {
    return (
        {
            filter => 'model_id='.$_[0]->_model->id,
        },
    );
}

sub invalid_param_sets {
}

sub _startup {
    my $self = shift;

    $self->_model->coolness('low');
    is($self->_model->coolness, 'low', 'Set coolness (input)');

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Names::Test;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Names';
}

sub valid_param_sets {
    return (
        {
            type_name => 'tester',
        },
    );
}

sub invalid_param_sets {
    return (
        {
            type_name => 'none',
        },
    );
}

sub startup : Tests(startup) {
    my $self = shift;
    
    no warnings 'once';
    no warnings 'redefine';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Update::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Update';
}

sub _valid_param_sets {
    return (
        { # remove single 'value'
            name => 'coolness',
            value => 'moderate',
        },
        { # remove single 'value'
            name => 'coolness',
            value => 'UNDEF',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to update an 'is_many' input
            name => 'friends',
            value => 'Watson',
        },
    );
}

###########################################################################

1;

=pod

=head1 Tests

=head1 Disclaimer

 Copyright (C) 2009 Washington University Genome Sequencing Center

 This script is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
 License for more details.

=head1 Author(s)

 Eddie Belter <ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

