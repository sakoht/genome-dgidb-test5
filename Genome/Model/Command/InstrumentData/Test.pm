
########################################################################

package Genome::Model::Command::InstrumentData::Test;

use strict;
use warnings;

#use base 'Genome::Utility::TestBase';


########################################################################

package Genome::Model::Command::InstrumentData::Assign::Test;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

require Genome::Model::Test;
use Test::More;

sub test_class {
    return 'Genome::Model::Command::InstrumentData::Assign';
}

sub params_for_test_class {
    return (
        model_id => $_[0]->model_id,
    );
}

sub required_params_for_class {
    return (qw/ model_id /);
}

sub model { return $_[0]->{_model}; }
sub model_id { return $_[0]->{_model}->id; }
sub instrument_data { return $_[0]->{_instrument_data}; }

sub create_mock_model_and_instrument_data : Test(startup => 2) {
    my $self = shift;

    # model
    my $model = Genome::Model::Test->create_basic_mock_model(type_name => 'tester');
    ok($model, 'Created mock model') or die;
    $self->{_model} = $model;

    # instrument data
    my @instrument_data = Genome::Model::Test->create_mock_sanger_instrument_data(4);
    is(scalar(@instrument_data), 4, 'Created 4 instrument data');
    $self->{_instrument_data} = \@instrument_data;

    $model->set_list('compatible_instrument_data', @instrument_data);
    
    return 1;
}

sub test_01_list : Tests {
    my $self = shift;

    ok($self->{_object}->execute, 'List instrument data');

    return 1;
}

sub test_02_assign_single_id : Tests {
    my $self = shift;

    my $assign = $self->test_class->create(
        $self->params_for_test_class,
        instrument_data_id => $self->instrument_data->[0]->id,
    );
    ok($assign->execute, 'Assign single instrument data');

    return 1;
}

sub test_03_assign_multiple_id : Tests {
    my $self = shift;

    my $assign = $self->test_class->create(
        $self->params_for_test_class,
        instrument_data_ids => join(
            ' ',
            $self->instrument_data->[1]->id,
            $self->instrument_data->[2]->id,
        ),
    );
    ok($assign->execute, 'Assign multiple instrument data');

    return 1;
}

sub test_04_assign_all : Tests {
    my $self = shift;

    my $assign = $self->test_class->create(
        $self->params_for_test_class,
        all => 1,
    );
    ok($assign->execute, 'Assign all instrument data');

    return 1;
}

sub test_05_multiple_actions : Tests {
    my $self = shift;

    my $assign = $self->test_class->create(
        all => 1,
        instrument_data_ids => '44 44',
    );
    ok(!$assign, 'Failed as expected - request multiple actions');

    return 1;
}

########################################################################

1;

#$HeadURL$
#$Id$
