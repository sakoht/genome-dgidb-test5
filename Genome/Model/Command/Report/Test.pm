package Genome::Model::Command::Report::TestBase;

use strict;
use warnings;

use Carp 'confess';
$SIG{__DIE__} = sub{ confess(@_) };

use base 'Genome::Utility::TestBase';

use Data::Dumper 'Dumper';
use Genome;
use Genome::Model::AmpliconAssembly::Test;
use Test::More;

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::AmpliconAssembly::Test->create_mock_model(use_test_dir => 1)
            or die "Can't create mock model\n";
    }

    return $self->{_model};
}

sub _build {
    my $self = shift;

    return $self->_model->latest_complete_build;
}

sub params_for_test_class {
    return (
        build_id => $_[0]->_build->build_id,
        $_[0]->_params_for_test_class,
    );
}

sub _params_for_test_class {
    return;
}

sub test_01_execute : Tests {
    my $self = shift;

    ok($self->{_object}->execute, 'Execute');
    
    return 1;
}

########################################################

package Genome::Model::Command::Report::GenerateTest;

use strict;
use warnings;

use base 'Genome::Model::Command::Report::TestBase';

use Data::Dumper 'Dumper';

sub test_class {
    return 'Genome::Model::Command::Report::Generate';
}

sub _params_for_test_class {
    return (
        report_name => 'Assembly Stats',
        directory => '/gscuser/ebelter/Desktop',
        force => 1,
    );
}

########################################################

package Genome::Model::Command::Report::GetDatasetTest;

use strict;
use warnings;

use base 'Genome::Model::Command::Report::TestBase';

use Data::Dumper 'Dumper';

sub test_class {
    return 'Genome::Model::Command::Report::GetDataset';
}

sub _params_for_test_class {
    return (
        report_name => 'Assembly Stats',
        dataset_name => 'stats',
        output_type => 'csv',
        #output_type => 'xml',
    );
}

1;

=pod

=head1 Name

ModuleTemplate

=head1 Synopsis

=head1 Usage

=head1 Methods

=head2 

=over

=item I<Synopsis>

=item I<Arguments>

=item I<Returns>

=back

=head1 See Also

=head1 Disclaimer

Copyright (C) 2005 - 2009 Genome Center at Washington University in St. Louis

This module is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> I<ebelter@genome.wustl.edu>

=cut

#$HeadURL$
#$Id$

