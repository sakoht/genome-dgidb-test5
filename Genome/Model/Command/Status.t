#!/gsc/bin/perl

use strict;
use warnings;

use File::Path;
use Test::More tests => 3;

use above 'Genome';

my $model_id = '2771359026';

my $model_status = Genome::Model::Command::Status->create(genome_model_id=>$model_id, display_output=>0);

ok($model_status);

my $rv = $model_status->execute;

is($rv, 1, 'Testing for successful execution.  Expecting 1.  Got: '.$rv);

my $length_test = 0;

my $xml = $model_status->xml();

if (length($xml) > 3000 ) {
    $length_test = 1 ;
} 

is($length_test,1,'Testing success: Expecting a longish XML string (>3000 chars). Got a string of length: '.length($xml));

