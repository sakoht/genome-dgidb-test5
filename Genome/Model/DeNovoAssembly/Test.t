#!/gsc/bin/perl

use strict;
use warnings;

use above "Genome";

use Data::Dumper 'Dumper';
use Test::More;

use_ok('Genome::Model::DeNovoAssembly::Test') or die;

my $rv;

#< PP >#
# fail - no seq platform
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model();
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - get w/o seq platform',
);
# fail - no assembler
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/o assembler',
);
# fail - invalid seq platform/assembler combo
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        assembler_name => 'consed',
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/ invalid seq paltform/assembler combo',
);

# valid 
my $model = Genome::Model::DeNovoAssembly::Test->get_mock_model(
    assembler_name => 'velvet',
    sequencing_platform => 'solexa',
);
ok($model, 'Got mock DNA model') or die;
my @inst_data = $model->instrument_data;
ok(@inst_data, 'Model instrument data');
ok(-s $inst_data[0]->archive_path, 'Solexa archive path exists');

done_testing();
exit;

#$HeadURL$
#$Id$
