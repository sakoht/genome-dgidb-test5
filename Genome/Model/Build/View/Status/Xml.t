#!/usr/bin/env perl
use strict;
use warnings;

use above "Genome";
use Genome::Model::Build::View::Status::Xml; 

use Test::More tests => 4;

# TODO: use one of the test builds
my $subject = Genome::Model::Build->get(101289765);
ok($subject, "found expected build subject") or die "test cannot continue...";

my $view_obj = Genome::Model::Build::View::Status::Xml->create(subject_id => 101289765, use_lsf_file => 1); 
ok($view_obj, "created a view") or die "test cannot continue...";

my $xml = $view_obj->_generate_content();
ok($xml, "view returns XML") or die "test cannot continue...";

my @diff =
    grep { $_ !~ /generated-at/ }
    grep { /\w/ }
    Genome::Utility::FileSystem->diff_file_vs_text(__FILE__ . '.expected',$xml);

is("@diff","","XML has no differences from expected value");

