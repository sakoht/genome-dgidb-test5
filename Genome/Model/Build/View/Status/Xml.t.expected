<?xml version="1.0"?>
<build-status generated-at="2010-04-28 16:14:50">
  <build model-name="ssmith-test2-realsample-realpp" model-id="2851407278" common-name="AML51" build-id="101289765" status="Succeeded" kilobytes-requested="4984448" data-directory="/gscmnt/sata427/info/model_data/2851407278/build101289765" lsf-job-id="966891" output-log="/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289767.out" error-log="/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289767.err">
    <model id="2851407278" type="Genome::Model::ReferenceAlignment">
      <aspect name="name">
        <value>ssmith-test2-realsample-realpp</value>
      </aspect>
    </model>
    <workflow instance-id="1753513" instance-status="done" id="1753513" type="Workflow::Model::Instance"/>
    <stages processing_profile="bwa0.5.5 and samtools r453 and picard r107" processing_profile_type="reference alignment">
      <stage value="alignment">
        <command_classes>
          <command_class value="Genome::Model::Event::Build::ReferenceAlignment::AlignReads">
            <events>
              <event id="101289768" command_class="Genome::Model::Event::Build::ReferenceAlignment::AlignReads::Bwa">
                <event_status>Succeeded</event_status>
                <instance_id>1753519</instance_id>
                <execution_count>1</execution_count>
                <instance_status>done</instance_status>
                <lsf_job_id>966892</lsf_job_id>
                <lsf_job_status>UNAVAILABLE</lsf_job_status>
                <date_scheduled>2010-02-28 22:12:06</date_scheduled>
                <date_completed>2010-02-28 22:36:11</date_completed>
                <elapsed_time>24:05</elapsed_time>
                <instrument_data_id>2849448911</instrument_data_id>
                <output_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/solexa/100109_HWUSI-EAS1605_100737092_61DTD/101289768.out</output_log_file>
                <error_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/solexa/100109_HWUSI-EAS1605_100737092_61DTD/101289768.err</error_log_file>
                <alignment_directory>/gscmnt/sata912/info/alignment_data/bwa0_5_5/cd6fc20ad94e451c80345e7544d30cc7/NCBI-human-build36/100109_HWUSI-EAS1605_100737092_61DTD/1_2849448911</alignment_directory>
              </event>
            </events>
          </command_class>
        </command_classes>
        <operating_on>
          <object value="instrument_data">
            <instrument_data id="2849448911">
              <project_name>AML Whole Genome Sequencing</project_name>
              <sample_name>H_KA-943309-S.18828</sample_name>
              <run_name>100109_HWUSI-EAS1605_100737092_61DTD</run_name>
              <flow_cell_id>61DTD</flow_cell_id>
              <read_length>100</read_length>
              <library_name>H_KA-943309-S.18828-lib1</library_name>
              <library_id>2847322310</library_id>
              <lane>1</lane>
              <subset_name>1</subset_name>
              <seq_id>2849448911</seq_id>
              <run_type>Paired</run_type>
              <gerald_directory>/gscmnt/sata807/production/100109_HWUSI-EAS1605_100737092_61DTD/Data/Intensities/BaseCalls/GERALD_100829249_1263920180_lims</gerald_directory>
            </instrument_data>
          </object>
        </operating_on>
      </stage>
      <stage value="deduplication">
        <command_classes>
          <command_class value="Genome::Model::Event::Build::ReferenceAlignment::DeduplicateLibraries">
            <events>
              <event id="101289769" command_class="Genome::Model::Event::Build::ReferenceAlignment::DeduplicateLibraries::Picard">
                <event_status>Succeeded</event_status>
                <instance_id>1753524</instance_id>
                <execution_count>1</execution_count>
                <instance_status>done</instance_status>
                <lsf_job_id>966931</lsf_job_id>
                <lsf_job_status>UNAVAILABLE</lsf_job_status>
                <date_scheduled>2010-02-28 22:36:33</date_scheduled>
                <date_completed>2010-02-28 23:14:48</date_completed>
                <elapsed_time>38:15</elapsed_time>
                <instrument_data_id/>
                <output_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289769.out</output_log_file>
                <error_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289769.err</error_log_file>
                <alignment_directory/>
              </event>
            </events>
          </command_class>
          <command_class value="Genome::Model::Event::Build::ReferenceAlignment::PostDedupReallocate">
            <events>
              <event id="101289770" command_class="Genome::Model::Event::Build::ReferenceAlignment::PostDedupReallocate">
                <event_status>Succeeded</event_status>
                <instance_id>1753525</instance_id>
                <execution_count>1</execution_count>
                <instance_status>done</instance_status>
                <lsf_job_id>966951</lsf_job_id>
                <lsf_job_status>UNAVAILABLE</lsf_job_status>
                <date_scheduled>2010-02-28 23:15:02</date_scheduled>
                <date_completed>2010-02-28 23:15:10</date_completed>
                <elapsed_time>00:08</elapsed_time>
                <instrument_data_id/>
                <output_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289770.out</output_log_file>
                <error_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289770.err</error_log_file>
                <alignment_directory/>
              </event>
            </events>
          </command_class>
        </command_classes>
        <operating_on>
          <object value="all_sequences"/>
        </operating_on>
      </stage>
      <stage value="variant_detection">
        <command_classes>
          <command_class value="Genome::Model::Event::Build::ReferenceAlignment::UpdateGenotype">
            <events>
              <event id="101289771" command_class="Genome::Model::Event::Build::ReferenceAlignment::UpdateGenotype::Samtools">
                <event_status>Succeeded</event_status>
                <instance_id>1753530</instance_id>
                <execution_count>1</execution_count>
                <instance_status>done</instance_status>
                <lsf_job_id>966953</lsf_job_id>
                <lsf_job_status>UNAVAILABLE</lsf_job_status>
                <date_scheduled>2010-02-28 23:15:26</date_scheduled>
                <date_completed>2010-02-28 23:15:27</date_completed>
                <elapsed_time>00:01</elapsed_time>
                <instrument_data_id/>
                <output_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/all_sequences/101289771.out</output_log_file>
                <error_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/all_sequences/101289771.err</error_log_file>
                <alignment_directory/>
              </event>
            </events>
          </command_class>
          <command_class value="Genome::Model::Event::Build::ReferenceAlignment::FindVariations">
            <events>
              <event id="101289772" command_class="Genome::Model::Event::Build::ReferenceAlignment::FindVariations::Samtools">
                <event_status>Succeeded</event_status>
                <instance_id>1753531</instance_id>
                <execution_count>1</execution_count>
                <instance_status>done</instance_status>
                <lsf_job_id>966970</lsf_job_id>
                <lsf_job_status>UNAVAILABLE</lsf_job_status>
                <date_scheduled>2010-02-28 23:15:43</date_scheduled>
                <date_completed>2010-03-01 00:01:48</date_completed>
                <elapsed_time>46:05</elapsed_time>
                <instrument_data_id/>
                <output_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/all_sequences/101289772.out</output_log_file>
                <error_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs/all_sequences/101289772.err</error_log_file>
                <alignment_directory/>
              </event>
            </events>
          </command_class>
        </command_classes>
        <operating_on>
          <object value="all_sequences"/>
        </operating_on>
      </stage>
      <stage value="generate_reports">
        <command_classes>
          <command_class value="Genome::Model::Event::Build::ReferenceAlignment::RunReports">
            <events>
              <event id="101289773" command_class="Genome::Model::Event::Build::ReferenceAlignment::RunReports">
                <event_status>Succeeded</event_status>
                <instance_id>1753536</instance_id>
                <execution_count>1</execution_count>
                <instance_status>done</instance_status>
                <lsf_job_id>966985</lsf_job_id>
                <lsf_job_status>UNAVAILABLE</lsf_job_status>
                <date_scheduled>2010-03-01 00:02:00</date_scheduled>
                <date_completed>2010-03-01 07:14:30</date_completed>
                <elapsed_time>07:12:30</elapsed_time>
                <instrument_data_id/>
                <output_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289773.out</output_log_file>
                <error_log_file>/gscmnt/sata427/info/model_data/2851407278/build101289765/logs//101289773.err</error_log_file>
                <alignment_directory/>
              </event>
            </events>
          </command_class>
        </command_classes>
        <operating_on>
          <object value="all_sequences"/>
        </operating_on>
      </stage>
    </stages>
    <reports directory="/gscmnt/sata427/info/model_data/2851407278/build101289765/reports/">
      <report name="Build Initialized" subdirectory="Build_Initialized"/>
      <report name="Build Succeeded" subdirectory="Build_Succeeded"/>
      <report name="dbSNP Concordance" subdirectory="dbSNP_Concordance"/>
      <report name="Input_Base_Counts" subdirectory="Input_Base_Counts"/>
      <report name="Mapcheck" subdirectory="Mapcheck"/>
      <report name="Summary" subdirectory="Summary"/>
    </reports>
  </build>
</build-status>
